awx_collection/test/awx/conftest.py

```diff
@@ -111,6 +111,86 @@ def mock_load_params(self):
+                elif isinstance(kwargs['params'], list):
+                    for k, v in kwargs['params']:
+                        kwargs_copy['data'][k] = v
+
+            # make request
+            rf = _request(method.lower())
+            django_response = rf(url, user=request_user, expect=None, **kwargs_copy)
+
+            # requests library response object is different from the Django response, but they are the same concept
+            # this converts the Django response object into a requests response object for consumption
+            resp = Response()
+            py_data = django_response.data
+            sanitize_dict(py_data)
+            resp._content = bytes(json.dumps(django_response.data), encoding='utf8')
+            resp.status_code = django_response.status_code
+
+            if request.config.getoption('verbose') > 0:
+                logger.info(
+                    '%s %s by %s, code:%s',
+                    method, '/api/' + url.split('/api/')[1],
+                    request_user.username, resp.status_code
+                )
+
+            return resp
+
+        def new_open(self, method, url, **kwargs):
+            r = new_request(self, method, url, **kwargs)
+            return mock.MagicMock(read=mock.MagicMock(return_value=r._content), status=r.status_code)
+
+        stdout_buffer = io.StringIO()
+        # Requies specific PYTHONPATH, see docs
+        # Note that a proper Ansiballz explosion of the modules will have an import path like:
+        # ansible_collections.awx.awx.plugins.modules.{}
+        # We should consider supporting that in the future
+        resource_module = importlib.import_module('plugins.modules.{0}'.format(module_name))
+
+        if not isinstance(module_params, dict):
+            raise RuntimeError('Module params must be dict, got {0}'.format(type(module_params)))
+
+        # Ansible params can be passed as an invocation argument or over stdin
+        # this short circuits within the AnsibleModule interface
+        def mock_load_params(self):
+            self.params = module_params
+
+        with mock.patch.object(resource_module.TowerModule, '_load_params', new=mock_load_params):
+            # Call the test utility (like a mock server) instead of issuing HTTP requests
+            with mock.patch('ansible.module_utils.urls.Request.open', new=new_open):
+                with mock.patch('tower_cli.api.Session.request', new=new_request):
+                    # Ansible modules return data to the mothership over stdout
+                    with redirect_stdout(stdout_buffer):
+                        try:
+                            resource_module.main()
+                        except SystemExit:
+                            pass  # A system exit indicates successful execution
+
+        module_stdout = stdout_buffer.getvalue().strip()
+        result = json.loads(module_stdout)
+        return result
+
+    return rf
```

Comment:
since the code diff with the other fixture is so small, I would like to combine the two before merging this